src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -935,9 +935,11 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 		}
 
 		// Open in target groups
-		const result: Promise<IEditorPane | null>[] = [];
+		const result: Promise<IEditorPane | undefined>[] = [];
 		mapGroupToEditors.forEach((editorsWithOptions, group) => {
-			result.push(group.openEditors(editorsWithOptions));
+			for (const editor of editorsWithOptions) {
```

Comment:
I think we have to be careful here to not break the semantics of `openEditors` vs `openEditor`. There is specific knowledge implemented inside an editor group for how to efficiently open multiple editors at once:

https://github.com/microsoft/vscode/blob/38f2172759e730a5e50684406c316ae2c40f279b/src/vs/workbench/browser/parts/editor/editorGroupView.ts#L1100-L1127

Can we revisit this choice? If this is about figuring out the right override, then I think the override part needs to happen before when we create editor inputs.

Reply:
Yes, I agree that this choice probably wasn't the best one. There's a lot of issues that arise due to the override not being respected in the openEditors flow so I was trying to think of the best way to do that here